Tests/GRPCCodeGenTests/Internal/Translator/SnippetBasedTranslatorTests.swift

```diff
@@ -0,0 +1,487 @@
+    let serviceB = ServiceDescriptor(
+      documentation: "Documentation for ServiceA",
+      name: "BService",
+      namespace: "",
+      methods: []
+    )
+
+    let serviceA = ServiceDescriptor(
+      documentation: "Documentation for ServiceA",
+      name: "AService",
+      namespace: "",
+      methods: []
+    )
+
+    let expectedSwift =
+      """
+      enum AService {
+          static let methods: [MethodDescriptor] = []
+          typealias StreamingServiceProtocol = AServiceServiceStreamingProtocol
+          typealias ServiceProtocol = AServiceServiceProtocol
+      }
+      enum BService {
+          static let methods: [MethodDescriptor] = []
+          typealias StreamingServiceProtocol = BServiceServiceStreamingProtocol
+          typealias ServiceProtocol = BServiceServiceProtocol
+      }
+      """
+
+    try self.assertTypealiasTranslation(
+      codeGenerationRequest: self.makeCodeGenerationRequest(services: [serviceB, serviceA]),
+      expectedSwift: expectedSwift
+    )
+  }
+
+  func testTypealiasTranslatorNamespaceAlphabeticalOrder() throws {
+    let serviceB = ServiceDescriptor(
+      documentation: "Documentation for BService",
+      name: "BService",
+      namespace: "bnamespace",
+      methods: []
+    )
+
+    let serviceA = ServiceDescriptor(
+      documentation: "Documentation for AService",
+      name: "AService",
+      namespace: "anamespace",
+      methods: []
+    )
+
+    let expectedSwift =
+      """
+      enum anamespace {
+          enum AService {
+              static let methods: [MethodDescriptor] = []
+              typealias StreamingServiceProtocol = anamespace_AServiceServiceStreamingProtocol
+              typealias ServiceProtocol = anamespace_AServiceServiceProtocol
+          }
+      }
+      enum bnamespace {
+          enum BService {
+              static let methods: [MethodDescriptor] = []
+              typealias StreamingServiceProtocol = bnamespace_BServiceServiceStreamingProtocol
+              typealias ServiceProtocol = bnamespace_BServiceServiceProtocol
+          }
+      }
+      """
+
+    try self.assertTypealiasTranslation(
+      codeGenerationRequest: self.makeCodeGenerationRequest(services: [serviceB, serviceA]),
+      expectedSwift: expectedSwift
+    )
+  }
+
+  func testTypealiasTranslatorNamespaceNoNamespaceOrder() throws {
+    let serviceA = ServiceDescriptor(
+      documentation: "Documentation for AService",
+      name: "AService",
+      namespace: "anamespace",
+      methods: []
+    )
+    let serviceB = ServiceDescriptor(
+      documentation: "Documentation for BService",
+      name: "BService",
+      namespace: "",
+      methods: []
+    )
+    let expectedSwift =
+      """
+      enum BService {
+          static let methods: [MethodDescriptor] = []
+          typealias StreamingServiceProtocol = BServiceServiceStreamingProtocol
+          typealias ServiceProtocol = BServiceServiceProtocol
+      }
+      enum anamespace {
```

Comment:
This highlights another potential issue: a service with no namespace can't have the same name as another namespace in the request. We'll need a test for this too.